### DISCUSSION TITLE
Question about computing resources in topo step

### DISCUSSION THREAD

**Discussion by fdelgadodelapuente** [isce-framework/isce2#412] [General]
Dear ISCE developers,

I've noticed that the topo step in stripmapApp and topsApp usually consumes a huge amount of computing resources in both Mac and Linux machines. topo uses so much resources, such that if I run another interferogram with ISCE, the computing time increases exponentially for the two pairs. Why is topo so resource demanding?

Francisco

**Comment by piyushrpt** [comment]
If you want to run multiple isce jobs, its up to you control number of threads available to each job using OMP_NUM_THREADS. If you don't do that - these jobs will just compete for all available resources. topo is demanding because it is searching 3D geometry space to determine which projects to a 2D pixel (az, rng) - it is searching a higher dimensional solution space.

**Reply by afpoley** [reply]
Hello, Is there any info on how to set the OMP_NUM_THREADS option? Thank you.

**Reply by afpoley** [reply]
I found the solution by setting OMP_NUM_THREADS as an environmental variable before running worked.

**Comment by fdelgadodelapuente** [comment]
Thanks for the explanation Piyush
